cdap-support-bundle/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -102,13 +125,99 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
+    // Ensure the user is authorized to list support bundle statuses
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.LIST);
+
+    List<SupportBundleOperationStatus> bundleOperationStatusList = bundleGenerator.getAllBundleStatus();
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(bundleOperationStatusList));
+  }
+
+  /**
+   * Get the specific bundle status from status.json by using uuid.
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws NotFoundException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundles/{uuid}/status")
+  public void getSupportBundleStatus(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws IOException, NotFoundException {
+    SupportBundleEntityId bundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(bundleEntityId, StandardPermission.GET);
+    SupportBundleOperationStatus bundleOperationStatus = bundleGenerator.getBundle(uuid);
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(bundleOperationStatus));
+  }
+
+  /**
+   * Delete the specific bundle by using the uuid
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws NotFoundException is thrown if status.json is not found
+   */
+  @DELETE
+  @Path("/support/bundles/{uuid}")
+  public void deleteSupportBundle(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws IOException, NotFoundException {
+    SupportBundleEntityId bundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(bundleEntityId, StandardPermission.DELETE);
+    bundleGenerator.deleteBundle(uuid);
+    responder.sendString(HttpResponseStatus.OK, String.format("Successfully deleted bundle %s", uuid));
+  }
+
+  /**
+   * Exports all selected files as a ZIP archive file.
+   *
+   * @param uuid the bundle id which is also the uuid fileJson body
+   */
+  @POST
+  @Path("/support/bundles/{uuid}")
+  public void downloadSupportBundle(FullHttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws BadRequestException, IOException, NotFoundException, NoSuchAlgorithmException {
+    SupportBundleEntityId bundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(bundleEntityId, StandardPermission.GET);
+    String requestContent = request.content().toString(StandardCharsets.UTF_8);
+    File tempDir = new File(cConf.get(Constants.SupportBundle.SUPPORT_BUNDLE_TEMP_DIR)).getAbsoluteFile();
+    DirUtils.mkdirs(tempDir);
+    java.nio.file.Path tmpPath = Files.createTempFile(tempDir.toPath(), "support-bundle", ".zip");
+    String digestHeader = "";
+    SupportBundleRequestFileList bundleRequestFileList;
+    try {
+      if (requestContent == null || requestContent.length() == 0) {
+        bundleRequestFileList = new SupportBundleRequestFileList(Collections.singletonList(""));
```

Comment:
shouldn't this be an empty list instead of a list with a single element??

Reply:
When the requested file list is empty, what does it mean? Does it mean downloading all files?

Reply:
yes when the request file is empty it means downloading all files